what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt

```diff
@@ -5,93 +5,221 @@ import android.content.ClipboardManager
+                this.putExtra(Intent.EXTRA_TEXT, stackTrace)
+                this.type = "text/plain"
             }
-        }
 
-        binding.shareStacktrace.apply {
-            setOnClickListener {
-                val sendIntent: Intent = Intent().apply {
-                    this.action = Intent.ACTION_SEND
-                    this.putExtra(Intent.EXTRA_TEXT, stackTrace)
-                    this.type = "text/plain"
-                }
+            val shareIntent = Intent.createChooser(sendIntent, null)
+            startActivity(shareIntent)
+        }
 
-                val shareIntent = Intent.createChooser(sendIntent, null)
-                startActivity(shareIntent)
+    @Composable
+    fun LaunchAppButton() =
+        ImageButton(
+            stringResource(R.string.restart_application),
+            R.drawable.ic_baseline_refresh_24
+        ) {
+            packageManager.getLaunchIntentForPackage(applicationContext.packageName)?.let {
+                it.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+                startActivity(it)
             }
         }
 
-        binding.launchApplication.apply {
-            setOnClickListener {
-                context.packageManager.getLaunchIntentForPackage(applicationContext.packageName)?.let {
-                    it.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
-                    startActivity(it)
-                }
-            }
+    @Composable
+    fun ImageButton(textString: String, @DrawableRes iconResId: Int, doOnClick: () -> Unit) =
+        OutlinedButton(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 16.dp, top = 16.dp, end = 16.dp, bottom = 2.dp),
+            onClick = doOnClick
+        ) {
+            Icon(
+                asset = vectorResource(id = iconResId),
+                tint = colorResource(id = R.color.secondaryColor)
+            )
+            Text(
+                modifier = Modifier.fillMaxWidth(),
+                textAlign = TextAlign.Center,
+                text = textString.toUpperCase(Locale.getDefault()),
+                color = colors.onSurface
+            )
         }
+    //endregion
+
+    //region StackTrace
+    @Composable
+    fun StackTraceHeader() {
+        Text(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(top = 16.dp, start = 8.dp, end = 16.dp),
+            text = stringResource(R.string.stacktrace).toUpperCase(Locale.getDefault()),
+            style = typography.overline
+        )
     }
 
-    private inline fun snackbar(messageProvider: () -> Int) {
-        Snackbar.make(binding.nestedScrollRoot, messageProvider(), Snackbar.LENGTH_SHORT).show()
+    @Composable
+    fun StackTrace(stackTraceText: String) {
+        ScrollableColumn(
+            Modifier
+                .fillMaxWidth()
+                .padding(top = 4.dp, start = 16.dp, end = 16.dp)
+        ) {
```

Comment:
To replicate the original design, remove the `ScrollableColumn` wrapper here, and change the `Column` child of `Scaffold` to `ScrollableColumn` instead. Additionally, add a modifier to the `ScrollableRow` to fill the max available height.